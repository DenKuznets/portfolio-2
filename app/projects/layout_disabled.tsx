
import { ReactNode } from 'react';

const ProjectsLayout = ({ children }: { children: ReactNode }) => {
    return (
        <>
            layout
            {children}
        </>
    );
};

export default ProjectsLayout;
