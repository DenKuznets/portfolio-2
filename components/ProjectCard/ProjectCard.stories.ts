import type { Meta, StoryObj } from "@storybook/react";
import ProjectCard from "./ProjectCard";

const meta = {
  title: "ProjectCard",
  component: ProjectCard,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
  tags: ["autodocs"],
  parameters: {
    // More on how to position stories at: https://storybook.js.org/docs/react/configure/story-layout
    layout: "fullscreen",
  },
} satisfies Meta<typeof ProjectCard>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
    args: {
        projectId: 1
    }
};
