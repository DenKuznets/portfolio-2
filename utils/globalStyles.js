import { createGlobalStyle } from "styled-components";

import { colors } from "./js/colors";
import { screen } from "./js/screen";
import { fonts } from "./js/fonts";

const GlobalStyle = createGlobalStyle`

  /* Box sizing rules */
*,
*::before,
*::after {
  box-sizing: border-box;
}

/* Remove default margin */
body,
h1,
h2,
h3,
h4,
p,
figure,
blockquote,
dl,
dd {
  margin: 0;
}

/* Remove list styles on ul, ol elements with a list role, which suggests default styling will be removed */
ul[role='list'],
ol[role='list'] {
  list-style: none;
}


/* Set core body defaults */
body {
  min-height: 100vh;
  text-rendering: optimizeSpeed;
  line-height: 1.5;
}

/* A elements that don't have a class get default styles */
a:not([class]) {
  text-decoration-skip-ink: auto;
}

/* Make images easier to work with */
img,
picture {
  max-width: 100%;
  display: block;
}

/* Inherit fonts for inputs and buttons */
input,
button,
textarea,
select {
  font: inherit;
}

/* Remove all animations, transitions and smooth scroll for people that prefer not to see them */
@media (prefers-reduced-motion: reduce) {
  html:focus-within {
    scroll-behavior: auto;
  }

  *,
  *::before,
  *::after {
    animation-duration: 0.01ms !important;
    animation-iteration-count: 1 !important;
    transition-duration: 0.01ms !important;
    scroll-behavior: auto !important;
  }
}

${"" /* ================================END OF RESETS */}



h1,h2,h3,h4,h5,h6 {
  font-family: ${fonts.raleweay};
}

h1 {
    font-size: 4.5rem;
    line-height: 1.1;
    @media (max-width: ${screen.mobileL}) {
        font-size: 3rem;
    }
}

h2{
  font-size: 2.5em;
}

h3{
  font-size: 1.7rem;
}

p {
  font-size: 1rem;
  color: ${({ theme }) => theme.global.pTextColor}; 
    
    
    @media (max-width: ${screen.mobileL}) {
        font-size: 0.8rem;
    }
}

a{
  color: ${({ theme }) => theme.global.aTextColor};
  text-decoration: none;
  transition: all 0.2s;
  cursor: pointer;
  &:hover {
    opacity: 1;
    text-decoration: underline;
  color: ${({ theme }) => theme.global.aHoverTextColor};
    
  }
}

ul {
  list-style-type: none;
  padding: 0;
  margin: 0;
}

section {
    padding: 0 1em;
  	width: 100%;
}

img, .small-icon, svg {
    user-select: none;
    -moz-user-select: none;
    -khtml-user-select: none;
    -webkit-user-select: none;
    -o-user-select: none;
}

.social-icon-container{
	padding: 14px;
	display: flex;
	align-items: center;
	justify-content: center;
	border-radius: 50%;
	transition: all 0.3s ease;
	color: ${({ theme }) => theme.global.socialIconContainerColor};
	background-color: ${({ theme }) => theme.global.socialIconContainerBG};
	&:hover{
		transform: scale(1.2);
	}
}
`;

export default GlobalStyle;
