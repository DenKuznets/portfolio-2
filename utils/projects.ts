export interface Project {
    id: number;
    name: string;
    tech: string[];
    img: string;
    github: string;
    demo: string;
    description: string;
}

export const getProject = (id: number) =>
    projects.find((proj) => proj.id === id);

export interface techListType {
    react: string,
    nextjs: string,
    tailwind: string,
    ts: string,
    mui: string,
    redux: string,
    rest: string,
    unittest: string,
    storybook: string,
    sc: string,
    classic: string,
}

export const techList: techListType = {
    react: 'REACT',
    nextjs: 'NEXTJS',
    tailwind: 'TAILWIND',
    ts: 'TS',
    mui: 'MUI',
    redux: 'REDUX',
    rest: 'REST',
    unittest:"UNIT TESTING",
    storybook: 'STORYBOOK',
    sc: 'STYLED-COMPONENTS',
    classic:"HTML CSS JS"
};

export const projects: Project[] = [
    {
        id: 1,
        name: 'rock paper scissors',
        tech: [
            techList.react,
            techList.mui,
            techList.ts,
            techList.storybook,
            techList.unittest,
        ],
        img: 'rpc.webp',
        github: 'https://github.com/DenKuznets/rock-paper-scissors',
        demo: 'https://rock-paper-scissors-kappa-tawny.vercel.app/',
        description: `Игра "Камень Ножницы Бумага" по дизайну https://www.frontendmentor.io/challenges/rock-paper-scissors-game-pTgwgvgH`
    },
    {
        id: 2,
        name: 'коммерческий сайт',
        tech: [techList.classic],
        img: 'tes.webp',
        github: 'https://github.com/DenKuznets/tes.loc',
        demo: 'https://tul-energo-service.ru/',
        description: `Верстка лэндинга по заказу web-студии alex-media.ru по макету Figma с соблюдением дедлайна, pixel-perfect попаданием в адаптивные брейкпоинты. В сотрудничестве с дизайнером и backend-разработчиком (php, laravel, blade)`
    },
    {
        id: 3,
        name: 'тестовое задание',
        tech: [techList.react, techList.sc, techList.rest],
        img: 'test1.webp',
        github: 'https://github.com/DenKuznets/dls-test',
        demo: 'https://dls-test.vercel.app/',
        description: `Тестовое задание по макету Figma и ТЗ: 
                        1) Горизонтальное меню - sticky, исчезает после 200px прокрутки
                        2) Мобильное меню - плавное скрытие/открытие
                        3) Подменю - только CSS
                        4) Получение данных fetch()/axios
                        5) Отображение списка постов - flex
                        6) Поиск постов - по заголовкам и тексту
                        7) Полная информация о посте по клику на пост - modal window     
                        `
    },
    {
        id: 4,
        name: 'тестовое задание для web-студии',
        tech: [techList.react, techList.sc],
        img: 'webstudio.webp',
        github: 'https://github.com/DenKuznets/testovoe-alexmedia',
        demo: 'https://denkuznets.github.io/testovoe-alexmedia/',
        description: `Тестовое задание по макету Figma и ТЗ: 
                        1) Сделать липкое меню.
                        2) Мобильное и декстпное меню должно в html быть одним блоком. 
                        3) Добавить Анимации по своему усмотрению. Можно использовать сторонние библиотеки , но без jquery.(не обязательно)
                        4) Cделать popup и в форме предусмотреть валидацию, Пока не заполненны все поля кнопка должна быть неактивна.(не обязательно)
                        5) Ховеры на ссылки, по своему усмотрению.
                        6) Мобильная верстка`
    },

    {
        id: 5,
        name: 'портфолио',
        tech: [techList.react, techList.sc, techList.ts],
        img: 'portfolio.webp',
        github: 'https://github.com/DenKuznets/masha-portfolio',
        demo: 'http://davinchushka.vercel.app/',
        description: `шаблон портфолио по дизайну figma`
    },
    {
        id: 6,
        name: 'квиз',
        tech: [techList.react, techList.rest, techList.sc],
        img: 'quizz.webp',
        github: 'https://github.com/DenKuznets/quizzical-app',
        demo: 'https://denkuznets.github.io/quizzical-app/',
        description: `Пример использования API. По макету figma в рамках задания сайта scrimba.com, курс Learn React`
    },
    {
        id: 7,
        name: 'компонент диаграмма расходов',
        tech: [techList.react, techList.ts, techList.sc],
        img: 'component.webp',
        github: 'https://github.com/DenKuznets/Expenses-chart-component',
        demo: 'https://denkuznets.github.io/Expenses-chart-component/',
        description: `Основная сложность состояла в том, что бы придумать как реализовать отображение диаграммы. Кроме того, пробовал свои силы в написании компонентов react с использованием typescript. Задание с сайта frontendmentor.io`
    },
    {
        id: 8,
        name: 'коммерческий сайт',
        tech: [techList.classic],
        img: 'tehavto.webp',
        github: 'https://github.com/DenKuznets/teh-avto.loc',
        demo: 'https://avtodom-techauto.ru/',
        description: `Верстка лэндинга по заказу web-студии alex-media.ru по макету Figma с соблюдением дедлайна, pixel-perfect попаданием в адаптивные брейкпоинты. В сотрудничестве с дизайнером и backend-разработчиком (php, laravel, blade)`
    },
    {
        id: 9,
        name: 'коммерческий сайт',
        tech: [techList.classic],
        img: 'bmk.webp',
        github: 'https://github.com/DenKuznets/bmk-new.loc',
        demo: 'https://bmk78.ru/',
        description: `Верстка лэндинга по заказу web-студии alex-media.ru по макету Figma с соблюдением дедлайна, pixel-perfect попаданием в адаптивные брейкпоинты. В сотрудничестве с дизайнером и backend-разработчиком (php, laravel, blade)`
    },
    {
        id: 10,
        name: 'коммерческий сайт',
        tech: [techList.classic],
        img: 'cleaning.webp',
        github: 'https://github.com/DenKuznets/cleaning-chistulya.loc',
        demo: 'https://chistylya.ru/',
        description: `Верстка лэндинга по заказу web-студии alex-media.ru по макету Figma с соблюдением дедлайна, pixel-perfect попаданием в адаптивные брейкпоинты. В сотрудничестве с дизайнером и backend-разработчиком (php, laravel, blade)`
    },
    {
        id: 11,
        name: 'коммерческий сайт',
        tech: [techList.classic],
        img: 'myp.webp',
        github: 'https://github.com/DenKuznets/myp-comfort',
        demo: 'https://mup-komfort.ru/',
        description: `Верстка лэндинга по заказу web-студии alex-media.ru по макету Figma с соблюдением дедлайна, pixel-perfect попаданием в адаптивные брейкпоинты. В сотрудничестве с дизайнером и backend-разработчиком (php, laravel, blade)`
    },
    {
        id: 12,
        name: 'мое портфолио',
        tech: [techList.nextjs, techList.ts, techList.tailwind, techList.storybook, techList.unittest],
        img: 'myportfolio.webp',
        github: 'https://github.com/DenKuznets/portfolio',
        demo: 'https://denkuznets.vercel.app/',
        description: `Этот сайт портфолио я разработал сам, вдохновляясь примерами таких работ как https://stefantopalovicdev.vercel.app/ , https://caferati.me/portfolio , https://brittanychiang.com/ и http://riccardozanutta.com/ . 
                    За основу дизайна использовал этот шаблон https://themeforest.net/item/orido-personal-portfolio-website-figma-template/screenshots/37527613?index=1`
    },

    {
        id: 13,
        name: "drag'n'drop на react",
        tech: [techList.react, techList.redux],
        img: 'dnd.webp',
        github: 'https://github.com/DenKuznets/react-dnd',
        demo: 'https://react-dnd-ten.vercel.app/',
        description: `В рамках тестового задания реализован react dnd. CSS внимание не уделялось, поскольку работа велась в условиях сильного ограничения по времени задания.  
                    ТЗ: 
                    Реализовать простой 2д планировщик.
                        1) Сделать блок со списком объектов на выбор(это могут быть 2д
                            фотографии столов, стульев, перегородок).
                        2) Сделать доску на которое можно добавлять эти объекты и
                            перемещать их мышкой. 
                        3) Должна быть возможность сохранения расстановки в
                                файл(сохранять id объекта и его координаты).
                        4) Также должна быть возможность импорта этого файла с
                                координатами и отображения сохраненной расстановки.
                        5) Ограничения на стек технологий:JavaScript React Redux Остальное на ваше усмотрение`
    }
];
