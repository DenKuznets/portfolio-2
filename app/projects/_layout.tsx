import { techList } from '@/utils/projects';
import Link from 'next/link';
import { ReactNode } from 'react';

const ProjectsPage = ({ params, children }: { params: { filter: string }, children: ReactNode }) => {
    console.log('layout',params.filter);
    
    return (
        <div className="pt-8">
            <div className="flex flex-wrap gap-x-4 gap-y-2">
                {Object.keys(techList).map((key) => (
                    <Link
                        className={`border p-1 cursor-pointer ${params.filter === techList[key] && 'bg-red-500'}`}
                        key={techList[key]}
                        href={`/projects/${key}`}
                    >
                        {techList[key]}
                    </Link>
                ))}
            </div>
            {children}
        </div>
    );
};

export default ProjectsPage;
