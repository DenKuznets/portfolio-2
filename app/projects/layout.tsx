import { techList } from '@/utils/projects';
import Link from 'next/link';
import { ReactNode } from 'react';

const ProjectsPage = ({ params, children }: { params: { filter: string }, children: ReactNode }) => {
    console.log(params.filter);
    
    return (
        <div className="pt-8">
            <div className="flex flex-wrap gap-x-4 gap-y-2">
                {Object.keys(techList).map((key) => (
                    <Link
                        className={`border p-1 cursor-pointer`}
                        key={techList[key]}
                        href={`/projects/${key}`}
                    >
                        {techList[key]}
                    </Link>
                ))}
            </div>
            {children}
        </div>
    );
};

export default ProjectsPage;
