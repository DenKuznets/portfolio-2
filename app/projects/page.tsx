import ProjectCard from '@/components/ProjectCard/ProjectCard';
import { projects as projectsList, techListType } from '@/utils/projects';
import Link from 'next/link';
import { techList } from '@/utils/projects';
import { SearchParams } from '@/types';
import text from '@/utils/text';

type Props = {
    searchParams: SearchParams;
};

const page = ({ searchParams }: Props) => {
    return (
        <div className="container px-2 md:px-4">
            <div className="mx-auto flex  max-w-3xl flex-wrap justify-center gap-x-4 gap-y-2 sm:pb-6">
                <Link
                    className={`cursor-pointer rounded border p-1 text-xs uppercase no-underline  sm:p-2 sm:text-sm md:text-base ${
                        !searchParams.filter && 'bg-customYellow '
                    }`}
                    href={text.links.project}
                >
                    {text.all}
                </Link>
                {Object.keys(techList).map((key) => (
                    <Link
                        className={`cursor-pointer rounded border p-1 text-xs sm:p-2 sm:text-sm  md:text-base ${
                            searchParams.filter === key &&
                            'bg-customYellow'
                        }`}
                        key={techList[key as keyof techListType]}
                        href={`/projects?filter=${key}`}
                    >
                        {techList[key as keyof techListType]}
                    </Link>
                ))}
            </div>

            <div className="max-w-[1440px] mx-auto flex flex-col flex-wrap gap-4 pt-4 sm:grid sm:grid-cols-2 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4">
                {projectsList
                    .filter((project) =>
                        searchParams.filter
                            ? project.tech.includes(
                                  techList[
                                      searchParams.filter as keyof techListType
                                  ]
                              )
                            : true
                    )
                    .map((project) => (
                        <ProjectCard key={project.id} projectId={project.id} />
                    ))}
            </div>
        </div>
    );
};

export default page;
