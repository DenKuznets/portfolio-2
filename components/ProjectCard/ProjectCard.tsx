import Link from 'next/link';
import React from 'react';
import { projects } from '../../utils/projects';
import Image from 'next/image';
import text from '@/utils/text';

export interface ProjectCardProps
    extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
    projectId: number;
}

const ProjectCard = ({ className, projectId }: ProjectCardProps) => {
    const project = projects.find((proj) => proj.id === projectId);

    return project ? (
        <Link
            className={`relative block min-h-[20rem] max-w-md min-w-[18rem] rounded border-2 border-zinc-900 text-white shadow [&>img]:hover:saturate-200 ${
                className ? className : ''
            }`}
            href={`${text.links.project}${projectId}`}
        >
            <Image
                className="object-cover blur-sm transition-all duration-300"
                fill
                src={`/images/works-preview/${project.img}`}
                alt={`${project.name} project image`}
            />
            <h4 className="absolute left-0 top-0 w-full bg-zinc-900/[.75] p-2 font-bold shadow-xl first-letter:capitalize">
                {project.name}
            </h4>
            <div className="absolute bottom-0 left-0 w-full">
                <ul className="mb-4 flex flex-wrap items-center gap-2 px-2 ">
                    {project.tech.map((obj) => (
                        <li
                            className="border-currentColor whitespace-nowrap rounded-md border border-dotted bg-zinc-900/[.75] px-2 py-1 text-xs md:text-base"
                            key={obj}
                        >
                            {obj}
                        </li>
                    ))}
                </ul>
                <div className="p-2 bg-zinc-900/[.75] font-sans font-bold">
                    {text.showmore}
                </div>
            </div>
        </Link>
    ) : (
        <div>project not found</div>
    );
};

export default ProjectCard;
