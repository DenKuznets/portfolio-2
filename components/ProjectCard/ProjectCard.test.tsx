import { render, screen } from '@testing-library/react';
import ProjectCard from './ProjectCard';
import '@testing-library/jest-dom';
import { getProject } from '@/utils/projects';

test('renders correctly', () => {
    const project = getProject(1);
    if (project) {
        render(<ProjectCard projectId={project.id} />);
        const link = screen.getByRole('link');
        const header = screen.getByRole('heading', { level: 4 });
        const img = screen.getByAltText(`${project.name} project image`);
        const listItems = screen.getAllByRole('listitem');

        expect(header).toHaveTextContent(project?.name);
        expect(link).toHaveAttribute('href', `/allworks/${project.id}`);
        expect(img).toBeInTheDocument();
        expect(listItems).toHaveLength(project.tech.length);
    }
});

test("renders ProjectCard unchanged", () => {
  const { container } = render(<ProjectCard projectId={1} />);
  expect(container).toMatchSnapshot();
});
