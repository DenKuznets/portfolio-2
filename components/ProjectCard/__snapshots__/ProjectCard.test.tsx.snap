// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProjectCard unchanged 1`] = `
<div>
  <a
    class="container [&>img]:hover:blur-xl shadow border-2 border-zinc-900 rounded block max-w-sm min-h-[30rem] relative "
    href="/allworks/1"
  >
    <img
      alt="ROCK PAPER SCISSORS project image"
      class="transition-all duration-300 blur-sm object-cover"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=%2Fimages%2Fworks-preview%2Frpc.webp&w=3840&q=75"
      srcset="/_next/image?url=%2Fimages%2Fworks-preview%2Frpc.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fworks-preview%2Frpc.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fworks-preview%2Frpc.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fworks-preview%2Frpc.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fworks-preview%2Frpc.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fworks-preview%2Frpc.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fworks-preview%2Frpc.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fworks-preview%2Frpc.webp&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
    <h4
      class="shadow-xl font-bold text-white p-4 absolute top-0 left-0 bg-zinc-900/[.75] w-full"
    >
      ROCK PAPER SCISSORS
    </h4>
    <ul
      class="absolute bottom-0 left-0 flex px-4 flex-wrap gap-3 items-center mb-4"
    >
      <li
        class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-white text-white bg-zinc-900/[.75] border"
      >
        react
      </li>
      <li
        class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-white text-white bg-zinc-900/[.75] border"
      >
        mui
      </li>
      <li
        class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-white text-white bg-zinc-900/[.75] border"
      >
        ts
      </li>
      <li
        class="whitespace-nowrap px-4 py-2 border-dotted rounded-md border-white text-white bg-zinc-900/[.75] border"
      >
        unit testing
      </li>
    </ul>
  </a>
</div>
`;
